packages/core/src/components/axes/toolbar.ts

```diff
@@ -0,0 +1,495 @@
+
+		// loading or empty state
+		if (isDataLoading || buttonList.length === 0) {
+			// put an empty rect to keep space unchanged
+			DOMUtils.appendOrSelect(container, "svg.toolbar-loading-spacer")
+				.append("rect")
+				.attr("height", Configuration.toolbar.height)
+				.attr("width", buttonSize * 3) // value doesn't matter but can't be empty
+				.attr("opacity", 0);
+		} else {
+			// render toolbar buttons sequentially
+			let buttonXPosition = 0;
+			buttonList.forEach((button) => {
+				// button container
+				const buttonContainer = DOMUtils.appendOrSelect(
+					container,
+					`svg.${button.id}`
+				).classed("toolbar-button", true);
+
+				// add button background rect
+				const buttonBackground = DOMUtils.appendOrSelect(
+					buttonContainer,
+					"rect.toolbar-button-background"
+				)
+					.attr("x", buttonXPosition)
+					.attr("y", 0)
+					.attr("width", buttonSize)
+					.attr("height", buttonSize);
+
+				const buttonIcon = DOMUtils.appendOrSelect(
+					buttonContainer,
+					"svg.toolbar-button-icon"
+				)
+					.attr(
+						"x",
+						buttonXPosition + Configuration.toolbar.iconPadding
+					)
+					.attr("y", Configuration.toolbar.iconPadding)
+					.attr("width", Configuration.toolbar.iconSize)
+					.attr("height", Configuration.toolbar.iconSize)
+					.attr("viewBox", "0 0 32 32")
+					.attr("role", Roles.IMG);
+
+				buttonIcon.html(button.iconSVGContent());
+				if (button.shouldBeDisabled()) {
+					buttonContainer
+						.classed("toolbar-button--disabled", true)
+						.classed("toolbar-button--focusable", false)
+						.attr("tabindex", -1)
+						.attr("role", null);
+					buttonIcon.classed("toolbar-button-icon--disabled", true);
+					buttonBackground.classed(
+						"toolbar-button-background--disabled",
+						true
+					);
+					buttonContainer.on("click", null).on("keyup", null);
+				} else {
+					buttonContainer
+						.classed("toolbar-button--disabled", false)
+						.classed("toolbar-button--focusable", true)
+						.attr("tabindex", 0)
+						.attr("role", Roles.BUTTON);
+					buttonIcon.classed("toolbar-button-icon--disabled", false);
+					buttonBackground.classed(
+						"toolbar-button-background--disabled",
+						false
+					);
+					buttonContainer
+						.on("click", button.clickFunction)
+						.on("keyup", () => {
+							if (
+								(event.key && event.key === "Enter") ||
+								event.key === " "
+							) {
+								event.preventDefault();
+
+								button.clickFunction();
+							}
+						});
+				}
+				buttonXPosition += buttonSize;
+			});
+
+			this.overflowButton = this.getContainerSVG().select(
+				"svg.toolbar-overflow-menu"
+			);
+
+			if (this.isOverflowMenuOpen()) {
+				// keep overflow menu displayed
+				this.showOverflowMenu(true);
+			}
+		}
+	}
+
+	isOverflowMenuOpen() {
+		return (
+			this.overflowMenu
+				.selectAll("ul.bx--overflow-menu-options--open")
+				.size() > 0
+		);
+	}
+
+	// show/hide overflow menu
+	showOverflowMenu(show: boolean) {
```

Comment:
@theiliad 
Yes, we use the boolean parameter to decide if we should show or hide the overflow menu.
I am not pretty sure what's your preference.
Do you like to have two separate functions (`showOverflowMenu` and `hideOverflowMenu`) ?
Or any other preferred function name ?

Reply:
My concern is the name of the function. Looking at the name `showOverflowMenu` I'd assume that calling it would `show` the overflow menu. Maybe we need to call it `updateOverflowMenu`